cis194/week1/stevemao/src/Homework.hs

```diff
@@ -0,0 +1,31 @@
+module Homework (toDigits, toDigitsRev, doubleEveryOther, sumDigits, validate) where
+
+toDigits :: Integer -> [Integer]
+toDigits a
+  | a <= 0  = []
+  | otherwise = toDigits (a `div` 10) ++ [a `mod` 10]
+
+toDigitsRev :: Integer -> [Integer]
+toDigitsRev = reverse . toDigits
+
+mapInd :: (a -> Int -> b) -> [a] -> [b]
+mapInd f l = zipWith f l [0..]
+
+doubleOrNot :: Integer -> Int -> Integer
+doubleOrNot a b
+  | b `mod` 2 == 0 = a * 2
+  | otherwise = a
+
+doubleEveryOther :: [Integer] -> [Integer]
+doubleEveryOther = mapInd doubleOrNot
+
+sumDigits :: [Integer] -> Integer
+sumDigits = sum . (concatMap toDigits)
```

Comment:
The normal function application in Haskell has the highest precedence and `(.)` is `infixr 9`
This means `concatMap toDigits` do not need any brackets around it.